File: lib/player.rb
class Player
  attr_accessor :name, :inventory, :location

  def initialize
    @location = Location.new(0,0)
    puts "What is your name?"
    @name = gets.chomp
    @inventory = []
  end

  def has_item(item_name)
    self.inventory.map(&:name).include?(item_name)
  end

  def add_to_inventory(item)
    self.inventory.push(item)
    puts "#{item.name.capitalize} has been added to your inventory."
  end

  def remove_from_inventory(item_name)
    if @inventory.delete_if { |i| i.name == item_name }
      puts "#{item_name} has been removed from your inventory"
    else
      puts "Sorry, you don't have #{item_name}"
    end
  end

  def print_inventory
    if @inventory.length > 0
      puts "\n"
      puts "INVENTORY"
      puts "========="
      @inventory.each do |item|
        puts "* #{item.name}: #{item.count}"
      end
      puts "\n"
    else
      puts "Sorry, your inventory is empty!"
      puts "Why not look around and see if you can find something to pick up!"
    end
  end
end

File: src/player.rs
use std::io;
use inventory_item::InventoryItem;
use location::Location;

#[derive(Debug, Eq, PartialEq)]
pub struct Player {
    pub name: String,
    inventory: Vec<InventoryItem>,
    pub location: Location,
}

impl Player {
    pub fn new(inventory: Vec<InventoryItem>, x: u64, y: u64) -> Player {
        // get user input an assign the input to `name`
        println!("What's your name?");

        let mut name = String::new();

        io::stdin()
            .read_line(&mut name)
            .expect("Could not read line");

        Player {
            name: name,
            inventory: inventory,
            location: Location { x: x, y: y },
        }
    }

    pub fn print_inventory(&self) {
        if self.inventory.is_empty() {
            println!("Oops! You don't have any items. Why not take a look around?");
        } else {
            println!("{:?}", self.inventory);
        }
    }

    pub fn add_to_inventory(&mut self, item: InventoryItem) {
        println!("{} has been added to your inventory!", item.name);
        self.inventory.push(item);
    }

    pub fn inventory(&self) -> &[InventoryItem] {
        &self.inventory[..]
    }

    pub fn inventory_mut(&mut self) -> &mut Vec<InventoryItem> {
        &mut self.inventory
    }

    pub fn location(&self) -> &Location {
        &self.location
    }

    pub fn location_mut(&mut self) -> &mut Location {
        &mut self.location
    }
}
